guides/micronaut-microservices-services-discover-eureka/bookcatalogue/java/src/main/java/example/micronaut/Book.java

```diff
@@ -1,24 +1,26 @@
 package example.micronaut;
 
+import io.micronaut.core.annotation.Creator;
 import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.NonNull;
 
 import javax.validation.constraints.NotBlank;
+import java.util.Objects;
 
 @Introspected
 public class Book {
 
     @NonNull
     @NotBlank
-    private String isbn;
+    private final String isbn;
+
     @NonNull
     @NotBlank
+    private final String name;
 
-    private String name;
-
-    public Book() {}
-
-    public Book(@NonNull @NotBlank String isbn, @NonNull @NotBlank String name) {
+    @Creator
```

Comment:
don't think @Creator is necessary. Am I missing anything?

If you want to specify it to educate the user about @Create I think we should add a callout with a callout snippet.